### PR TITLE
Add a maximum-attention hook to help with debugging boot issues

### PR DESCRIPTION
**Pull request by MayeulC**
The hook flashes every LED it can find, as well as the vibration motor.
More paths are probably necessary to support more devices and kernels.

This is intended as a first approach to #1217
Also somewhat related to #88 

The name comes from ollieparanoid on Matrix, IIRC.

### CODE REVIEW

File: aports/main/postmarketos-mkinitfs-hook-maximum-attention/00-maximum-attention.sh [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+
+# blink_leds takes a list of LEDs as parameters,
+# it iterates over every LED, and changes their value,
+# alternating between max_brightness and 0 every BLINK_INTERVAL
+blink_leds() {
+        state=false # false = off, true=on
+        while :
+        do
+
```
Comment (MayeulC) [inline aports/main/postmarketos-mkinitfs-hook-maximum-attention/00-maximum-attention.sh]: OK, there might be a stray whiteline here

**Comment by MartijnBraam** [comment]
Hmm it doesn't seem to work with the notification led on my hammerhead, but that led has a slightly more complicated sysfs interface. To turn that one do:

```bash
cat max_brightness > brightness
echo 100 0 > on_off_ms
# You have to write a different value to rgb_start for every change
echo 1 > rgb_start
```
You can also make the kernel blink the led instead of the script by setting on_off_ms to the blinking times

```bash
echo 2000 2000 > on_off_ms
```

I guess you can just check if the current led has a rgb_start file to do this procedure before entering the blinking loop

**Comment by MayeulC** [comment]
Thanks a lot for your review, @MartijnBraam. THere are a few things there (hence why I called it a first approach):
* The LCD backlight should flash as well (if applicable), but I am not sure whether I should add a new splash to make it more visible
* currently, the hook loops forever. I am not sure if it should continue the boot process or not, I think it is better this way, as it is just intended as a one-off to diagnose boot issues
* There seem to be a lot of different vendor interfaces out there. I didn't really want to support non-standard ones, but I guess it would make the porting process easier.

I didn't really understand the purpose of `rgb_start` here, though.

**Comment by drebrez** [comment]
@MayeulC when I get home I will try on the various devices I have and let you know the results

**Comment by MartijnBraam** [comment]
The backlight goes to full brightness and then turns of, that is probably a sideeffect of some fbdev code in the kernel but the led does get controlled once.

Also the rgb led is slightly weird but the `rgb_start` file is to make the new settings apply all at once since you have to change 2 (or more) settings and accros 3 led devices usually if you wan't to flash a non-primary color. 

I think this hook should support as many leds as possible since it is for debugging booting issues.

**Comment by drebrez** [comment]
Tested and here are the results:

| device         | display backlight | button backlight | front led | camera led | vibrator |
|----------------|-------------------|------------------|-----------|------------|----------|
| samsung-i9070  | NO                | YES              | -         | NO         | YES      |
| samsung-maguro | YES               | NO               | NO        | NO         | YES      |
| htc-bravo      | YES               | YES              | YES       | YES        | YES      |
| samsung-klte   | YES               | YES              | YES       | YES        | YES      |
| huawei-y530    | NO                | YES              | YES       | NO         | YES      |

File: aports/main/postmarketos-mkinitfs-hook-maximum-attention/00-maximum-attention.sh [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+BLINK_INTERVAL=2 # seconds
+VIBRATION_DURATION=400 #ms
+VIBRATION_INTERVAL=2 #s
+
+find_leds() {
+	find /sys -name "max_brightness" | xargs -I{} dirname {}
+}
+
+find_vibrator()
+{
```
Comment (ollieparanoid) [inline aports/main/postmarketos-mkinitfs-hook-maximum-attention/00-maximum-attention.sh]: How about `find_vibrator() {` for consistency?

File: aports/main/postmarketos-mkinitfs-hook-maximum-attention/00-maximum-attention.sh [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+			state=false
+		else
+			state=true
+		fi
+	done
+}
+
+# vibrate_loop vibrates each VIBRATION_INTERVAL for VIBRATION_DURATION
+# it takes a timed_device path to the vibrator as $1
+vibrate_loop()
```
Comment (ollieparanoid) [inline aports/main/postmarketos-mkinitfs-hook-maximum-attention/00-maximum-attention.sh]: same here

File: aports/main/postmarketos-mkinitfs-hook-maximum-attention/00-maximum-attention.sh (hunk 2 above)
Comment (MayeulC) [reply aports/main/postmarketos-mkinitfs-hook-maximum-attention/00-maximum-attention.sh]: Yes, you are right, I am just *soo* used to my usual C syntax that I tend to mix things up. Thanks for noticing!

**Comment by MayeulC** [comment]
@ollieparanoid, I fixed the formatting issue (thanks for noticing! I have C-style formatting hardcoded in my fingers).

I think this should get merged, I will (maybe with some others) think about a way to add more device in a way that's as generic as possible, but will probably be busy for a few days.

**Pull request merged**